src/datachain/lib/listing.py

```diff
@@ -195,5 +196,6 @@ def get_listing(
         list_path = f"{ds_name.strip('/').removeprefix(listing.name)}/{list_path}"
 
     ds_name = listing.name if listing else ds_name
+    ds_name = DatasetRecord.clean_name(ds_name)
```

Comment:
this logic with cleaning listing names seems to be off ... 

e.g. 

`ds_name, list_uri, list_path = parse_listing_uri(uri)`

probably returns "dirty" version?

but then

```
ls for ls in catalog.listings() if not ls.is_expired and ls.contains(ds_name)
```

will be comparing it with cleaned versions?

also, are there other places in code where we construct listing ds names (e.g. via `parse_listing_uri`)?